feat: drag X on mobile (#9)

MARCxGAMBIT opened the pull request.

samimsu commented:
After trying to remove a queen, instead of showing a blank square it shows an x. I believe I faced a similar issue when i tried to implement it myself.

https://github.com/user-attachments/assets/89980ac7-2e61-45cd-883e-99a709e4ba2c

MARCxGAMBIT commented:
@samimsu will take a look. Seems like I forgot to test some edge cases.

MARCxGAMBIT commented:
I updated my PR. I added more states to the Board class to keep track of the origin of the Pointer events. Click is now triggered in pointerUp and not pointerDown (and also in pointerEnter when dragging). TouchMove is not needed anymore. Interestingly it works even if the touchMove code is still active. Seems like this automatically cancels the pointerEnter events.  I added some comments in the code to explain the finer details. Let me know if this works.

samimsu commented:
Works great! Thanks a lot @MARCxGAMBIT

samimsu commented:
@all-contributors please add @MARCxGAMBIT for code
